lib/mapping/mapping-handler.js

```diff
@@ -0,0 +1,375 @@
+'use strict';
+
+const utils = require('../utils');
+const QueryGenerator = require('./query-generator');
+const ResultMapper = require('./result-mapper');
+const Result = require('./result');
+const Cache = require('./cache');
+const Tree = require('./tree');
+const ObjectSelector = require('./object-selector');
+const DocInfoAdapter = require('./doc-info-adapter');
+
+/**
+ * @ignore
+ */
+class MappingHandler {
+  /**
+   * @param {Client} client
+   * @param {ModelMappingInfo} mappingInfo
+   */
+  constructor(client, mappingInfo) {
+    this._client = client;
+    this._cache = {
```

Comment:
hm... yes, it can be misused... 
What do you think about including a warning when there are more than x number of items on each tree/map instance (e.g., 100)?

Reply:
That sounds like a good way to go about it.  Really if you have a large amount of cache entries it's likely you are creating your queries in a suboptimal way, so we should warn.   I think 100 is a good high watermark.

Reply:
I've added validation and included 2 unit tests for it.